lib/grape/validations/validators/coerce.rb

```diff
@@ -11,7 +11,12 @@ def validate_param!(attr_name, params)
         if valid_type?(new_value)
           params[attr_name] = new_value
         else
-          fail Grape::Exceptions::Validation, params: [@scope.full_name(attr_name)], message_key: :coerce
+          bad_value = new_value
+          if bad_value.is_a?(Types::InvalidValue) && !bad_value.message.nil?
```

Comment:
I am not in love with this. We're now explicitly checking what kind of exception we got. Any better suggestions?